spring-integration-core/src/main/java/org/springframework/integration/aggregator/AbstractAggregatingMessageGroupProcessor.java

```diff
@@ -64,11 +79,12 @@ public final Object processMessageGroup(MessageGroup group) {
 		Map<String, Object> headers = this.aggregateHeaders(group);
 		Object payload = this.aggregatePayloads(group, headers);
 		AbstractIntegrationMessageBuilder<?> builder;
+		MessageBuilderFactory messageBuilderFactory = getMessageBuilderFactory();
```

Comment:
We don't need to store in a local variable here; it's only used once on each branch of the `if`.